doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,337 @@
+#  Smart Switch Reboot High Level Design
+
+## Table of Contents ##
+
+- [Smart Switch Reboot Design](#smart-switch-reboot-design)
+  - [Table of Contents](#table-of-contents)
+  - [Revision](#revision)
+  - [Glossory](#glossary)
+  - [Overview](#overview)
+  - [Assumptions](#assumptions)
+  - [Requirements](#requirements)
+  - [Methods of Switch and DPU Reboot](#methods-of-switch-and-dpu-reboot)
+  - [DPU reboot sequence](#dpu-reboot-sequence)
+  - [Switch reboot sequence](#switch-reboot-sequence)
+  - [High Level Design](#high-level-design)
+    - [ModuleBase Class API](#modulebase-class-api)
+    - [ModuleBase Class new APIs](#modulebase-class-new-apis)
+    - [NPU platform.json](#npu-platformjson)
+    - [GNOI API implementation](#gnoi-api-implementation)
+    - [reboot CLI modifications](#reboot-cli-modifications)
+    - [reboot.py script modifications](#rebootpy-script-modifications)
+    - [Error handling and exception scenarios](#error-handling-and-exception-scenarios)
+  - [Test plan](#test-plan)
+
+## Revision ##
+
+| Rev | Date       | Author           | Change Description |
+| --- | ---------- | ---------------- | ------------------ |
+| 0.1 | 05/16/2024 | Vasundhara Volam | Initial version    |
+| 0.2 | 05/29/2024 | Vasundhara Volam | Update images and APIs |
+| 0.3 | 06/10/2024 | Vasundhara Volam | Minor changes based on discussion with the community |
+
+## Glossory ##
+
+| Term  | Meaning                                   |
+| ----- | ----------------------------------------- |
+| ASIC  | Application-Specific Integrated Circuit   |
+| DPU   | Data Processing Unit                      |
+| GNMI  | gRPC Network Management Interface         |
+| GNOI  | gRPC Network Operations Interface         |
+| NPU   | Network Processing Unit                   |
+| PCI-E | Peripheral Component Interconnect Express |
+
+## Overview ##
+
+Smart Switch aims to provide a full suite of network functionality, like traditional network devices, but with the flexibility
+and scalability of cloud-based services. It consists of one switch ASIC (NPU) and multiple DPUs. The DPU ASICs are only
+connected to the NPU, and all front panel ports are connected to the NPU. The DPU also connects to the SmartSwitch CPU via PCI-E
+interfaces, allowing the Switch CPU to control DPUs through these interfaces.
+
+Each DPU will have one internal management IP which is used for internal communications, such as Redis database and zmq. This
+internal communication is also used between NPU and DPU, and between DPUs.
+
+This document provides high level design of reboot sequence of a SmartSwitch with multiple DPUs and single DPU reboot sequence
+through GNOI API.
+
+## Assumptions ##
+
+Smart Switch supports only cold-reboot and does not support warm-reboot as of today.
+
+## Requirements ##
+
+1. NPU host is running GNMI service to communicate with DPU.
+2. DPU host is running GNMI server to listen to GNOI client requests.
```

Comment:
DPU host should run gNOI server to listen to gNOI client requests. GNMI server should be separate from gNOI server- gNMI server for each DPU runs on NPU

Reply:
Modified it. Thank you!